pyteal/ast/subroutine.py

```diff
@@ -374,15 +427,138 @@ def name(self) -> str:
         return self.subroutine.name()
 
     def type_of(self):
-        return self.subroutine.getDeclaration().type_of()
+        return self.subroutine.get_declaration().type_of()
 
     def has_return(self):
-        return self.subroutine.getDeclaration().has_return()
+        return self.subroutine.get_declaration().has_return()
 
 
 SubroutineFnWrapper.__module__ = "pyteal"
 
 
+class ABIReturnSubroutine:
```

Comment:
@ahangsu and reviewers:  Since `ABIReturnSubroutine` is primarily (as I understand it) intended for ARC-4 method definition, I'm wondering how folks feel about matching the name with the use case?

* For example, I can imagine naming it `ABIMethod` in hopes of making the intended usage clearer to PyTeal developers, and providing consistency with https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0004.md#methods.
* On the other hand, it's possible that we _prefer_ a more general name in case use cases change over time.
* My starting 2c:  
  * I'd name it for the use case we have today (something like `ABIMethod`) because I think it's easier to generalize later rather than become increase specificity later. 
  * I'll follow the majority opinion here including keeping as is.  Since name changes are difficult (due to backwards compatibility concerns), I raise the question to sanity check the rubric applied to choose a name.

Reply:
I agree that it's better to name it based on it's prominent use case. It may, in fact, provide more functionality, but that's not a hindrance in my opinion (e.g. python's `list` has additional methods that let you use it as a **Stack** and this is just fine). An additional benefit is that the more specific the name, the easier it is to grasp.

Upshot: I support renaming to `ABIMethod`

Reply:
I am `1 - 9/19` supporting renaming this to `ABIMethod`:
- length of `Subroutine` is 10
- length of `ABIReturnSubroutine` is 19
- length of `ABIMethod` is 9

SWE's finger-joint life matters! If we can reach to a consensus that we want to rename this, I will do it.

Reply:
🍾

Reply:
Had a verbal with @tzaffi, @ahangsu, @algoidurovic, and @jannotti to discuss naming and landed on keeping as is.  

`ABIMethod` implies the method is auto-registered for routing.  Since auto-registration is _not_ provided, the name doesn't fit well.

Resolving the thread because it feels like we reached agreement.  If folks that weren't in the discussion feel otherwise, feel welcomed to reopen.